File: src/adventofcode2018/Day01.java
package adventofcode2018;

import java.util.Arrays;
import java.util.Collections;
import java.util.HashSet;
import java.util.Set;

import utils.Input;

/**
 * https://adventofcode.com/2018/day/1
 */
public class Day01 {

    public static void main(String[] args) {

        int[] numbers = new Input(2018, "input01.txt")
                .strings().stream()
                .mapToInt(Integer::parseInt)
                .toArray();

        System.err.println(Arrays.stream(numbers).sum());

        int result = 0;
        Set<Integer> reached = new HashSet<>(Collections.singleton(0));
        for (int i=0; ; i++) {
            result += numbers[i % numbers.length];
            if (reached.contains(result)) break;
            reached.add(result);
        }
        System.err.println(result);
    }

}

File: src/Python/2019/Day01.py

from utils import Input, printResult

# https://adventofcode.com/2019/day/1

def calc_fuel(mass, rec):
    mass = mass // 3 - 2
    if rec and mass > 8:
        mass += calc_fuel(mass, rec)
    return mass

input = list(map(int, Input(2019, 1).lines()))

printResult(1, sum(map(lambda x: calc_fuel(x, False), input)))
printResult(2, sum(map(lambda x: calc_fuel(x, True), input)))
